libs/template/experimental/virtual-scrolling/src/lib/virtual-for.directive.ts

```diff
@@ -609,11 +608,14 @@ export class RxVirtualFor<T, U extends NgIterable<T> = NgIterable<T>>
       updateViewContext: this.updateViewContext.bind(this),
       templateCacheSize: this.templateCacheSize,
     });
-    this.render()
-      .pipe(takeUntil(this._destroy$))
-      .subscribe((v) => {
-        this._renderCallback?.next(v as U);
-      });
+    // let the scroll strategy initialize before
+    Promise.resolve().then(() => {
```

Comment:
What about using the primitive operation `queueMicrotask` instead?

Reply:
both are patched by zone and I'm using the `Promise` from `cdk/zone-less/browser`. Of course I could also use the unpatched `queueMicrotask`. But is there any difference besides it looks a little bit better?

Reply:
No big difference, it takes less memory as there's no object to hold, but nothing noticeable I guess. However it's easier to read.